File: behavioral/dart/command_structured/receivers.dart
class Light {
  void on() {
    print('Light is ON');
  }
  
  void off() {
    print('Light is OFF');
  }
}

class Stereo {
  void on() {
    print('Stereo is ON');
  }
  
  void off() {
    print('Stereo is OFF');
  }
  
  void setVolume(int level) {
    print('Stereo volume set to $level');
  }
}

class BankAccount {
  double _balance = 0;
  
  BankAccount([double initialBalance = 0]) : _balance = initialBalance;
  
  void deposit(double amount) {
    _balance += amount;
    print('Deposited \$$amount. New balance: \$$_balance');
  }
  
  void withdraw(double amount) {
    if (_balance >= amount) {
      _balance -= amount;
      print('Withdrew \$$amount. New balance: \$$_balance');
    } else {
      print('Insufficient funds');
    }
  }
  
  double getBalance() => _balance;
}

File: behavioral/typescript/command_structured/receivers.ts
export class Light {
  on(): void {
    console.log('Light is ON');
  }
  
  off(): void {
    console.log('Light is OFF');
  }
}

export class Stereo {
  on(): void {
    console.log('Stereo is ON');
  }
  
  off(): void {
    console.log('Stereo is OFF');
  }
  
  setVolume(level: number): void {
    console.log(`Stereo volume set to ${level}`);
  }
}

export class BankAccount {
  private balance: number;
  
  constructor(initialBalance: number = 0) {
    this.balance = initialBalance;
  }
  
  deposit(amount: number): void {
    this.balance += amount;
    console.log(`Deposited $${amount}. New balance: $${this.balance}`);
  }
  
  withdraw(amount: number): void {
    if (this.balance >= amount) {
      this.balance -= amount;
      console.log(`Withdrew $${amount}. New balance: $${this.balance}`);
    } else {
      console.log('Insufficient funds');
    }
  }
  
  getBalance(): number {
    return this.balance;
  }
}
